### PR TITLE
Adding functionality to search resources by filename

### PR DESCRIPTION
**Pull request by pt2302**
#### Pre-Flight checklist

- [X] Testing
  - [X] Code is tested
  - [X] Changes have been manually tested

#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-studio/issues/1474.

#### What's this PR do?
Adds search by filename to content pages in OCW Studio.

#### How should this be manually tested?
Change the title of a resource to something entirely different from the title by using the Edit Resource drawer, and confirm that it is possible to find the resource by searching for a partial filename (including extensions).

### CODE REVIEW

File: websites/views.py [hunk 1]
```diff
@@ -575,7 +575,9 @@ def get_queryset(self):
         if types:
             queryset = queryset.filter(type__in=types)
         if search:
-            queryset = queryset.filter(title__icontains=search)
+            queryset = queryset.filter(
+                Q(title__icontains=search) | Q(file__iregex=f"{search}([^/]+|$)")
```
Comment (mbertrand) [inline websites/views.py:579]: This still needs to be tweaked a bit to:
```
Q(file__iregex=f"{search}([^/]+$|$)
```

For example I have a site with `short_id=new-site-1` and a resource with `file=courses/new-site-1/biology.jpeg`

Searching for `new` with current code returns the above resource (and all other resources for the site, they all have the same path).  With the suggested tweak, it won't be returned, but a resource with `file=courses/new-site-1/biology-new-photo.jpeg` will be.

File: websites/views_test.py [hunk 2]
```diff
@@ -702,6 +707,7 @@ def test_websites_content_list(  # pylint: disable=too-many-locals
             website=website,
             title=f"some TEXT{num} here for a case insensitive search",
             metadata={"resourcetype": "Image" if num % 2 == 0 else "Video"},
+            file=f"myFile{num+1}.png",
```
Comment (mbertrand) [inline websites/views_test.py:710]: Include a path here to make sure it is being filtered out from search results:
```
file=f"courses/my-test-site/myFile{num+1}.png",
```

File: websites/views_test.py [hunk 3]
```diff
@@ -680,6 +680,11 @@ def test_website_starters_site_configs_exception(mocker, drf_client):
         ["", "page", "", 5],
         ["", "", "text2", 1],
         ["", "", "", 6],
+        ["", "", "myfile3", 1],
+        ["", "", "myfile", 5],
+        ["", "", ".png", 5],
+        ["", "", ".p", 5],
+        ["", "", ".pdf", 0],
```
Comment (mbertrand) [inline websites/views_test.py:687]: Add some tests to make sure a search matching only the path part of a resource file won't be returned (see comment below):

```python
["", "", "test", 0],
["", "", "courses", 0],
```

**Review by mbertrand — CHANGES_REQUESTED** [review]

File: websites/views.py (hunk 1 above)
Comment (pt2302) [reply websites/views.py]: Updated in 929092e405c3630d744d9e3e0b96e1e9129fb513.

File: websites/views_test.py (hunk 2 above)
Comment (pt2302) [reply websites/views_test.py]: Updated in 929092e405c3630d744d9e3e0b96e1e9129fb513.

**Review by mbertrand — APPROVED** [review]
👍

**Pull request merged**